### PR TITLE
Remove unused payee rules feature

### LINKED ISSUE #615: Remove the “payee rules” feature
I looked through the code, and the `payee_rules` table appears to be unused. I think it would be safe to remove the table and associated code, although I would like @jlongster to weigh in on if it has some purpose that I’ve missed in my search.

### PR DESCRIPTION
**Pull request by j-f1**
Fixes #615. I would appreciate double-checking that I didn’t accidentally delete anything that is important.

Since I’m removing the related API methods, this is technically a breaking change (even if people would have no reason to remove this stuff), so we should probably do a major release of the API package.

### CODE REVIEW

File: packages/loot-core/src/server/db/index.ts [hunk 1]
```diff
@@ -454,10 +453,6 @@ export async function deletePayee(payee) {
   //   mappings.map(m => update('payee_mapping', { id: m.id, targetId: null }))
   // );
```
Comment (MatissJanis) [inline packages/loot-core/src/server/db/index.ts:454]: **🥜 nitpick:** ‏IMO worth deleting this too (but feel free to ignore for your PR as this is scope creep)

File: packages/api/methods.js [hunk 2]
```diff
@@ -149,19 +149,3 @@ export function updatePayee(id, fields) {
...
-
-export function getPayeeRules(payeeId) {
-  return send('api/payee-rules-get', { payeeId });
-}
-
-export function createPayeeRule(payeeId, rule) {
-  return send('api/payee-rule-create', { payee_id: payeeId, rule });
-}
-
-export function updatePayeeRule(id, fields) {
```
Comment (MatissJanis) [inline packages/api/methods.js:161]: There's a few more of these in `packages/loot-core/src/server/db/index.ts`

Comment (j-f1) [reply packages/api/methods.js:161]: I don’t see any?

File: packages/loot-core/src/server/db/index.ts (hunk 1 above)
Comment (j-f1) [reply packages/loot-core/src/server/db/index.ts:454]: Gonna hold off for now

File: packages/api/methods.js (hunk 2 above)
Comment (MatissJanis) [reply packages/api/methods.js:161]: https://github.com/actualbudget/actual/blob/master/packages/loot-core/src/server/db/index.ts#L545

Comment (MatissJanis) [reply packages/api/methods.js:161]: Honestly.. I am blind. You already removed these.

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**